### DISCUSSION TITLE
Issues after migration 1.2.4 -> 2.3.1

### DISCUSSION THREAD

**Discussion by mvolf** [quarkiverse/quarkus-quinoa#611] [Q&A] [answered]
I just migrated my React project from quinoa 1.2.4 to 2.3.1 (using Quarkus 3.6.4). After the migration I can't access any of the APIs from the application while in dev (live-coding) mode.
When I kill my application I can see errors like the following (for the APIs I tried to access):

`Quinoa failed to forward request '/api/health', see logs. [Error Occurred After Shutdown]: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:3003`

`Quinoa failed to forward request '/api/test', see logs. [Error Occurred After Shutdown]: java.net.SocketException: Connection reset`

Quinoa is set to use 3003 port and my start script is defined as follows:

`"start": "export PORT=3003 && react-scripts start",`

The only change in the configuration that was needed (as far as I've seen) when migrating from 1.2.4 -> 2.3.1 was changing `quarkus.quinoa.frozen-lockfile=false` to `quarkus.quinoa.ci=false`.
Does anyone have any clue what could be causing this issue?

**Comment by melloware** [comment]
@mvolf can you post all your log from the startup?  I wonder if its Quinoa now auto-detecting a framework and overriding your settings?

**Reply by ia3andy** [reply]
Well it should not override user settings, that would be a bug.

**Reply by mvolf** [reply]
Thanks for the reply @melloware . Of course, here are the Quinoa related log lines from the startup:

```
2024-01-09 07:56:31,197 INFO  [io.qua.qui.dep.fra.FrameworkType] (build-54) Quinoa detected 'REACT' frameworkType from package.json file.
2024-01-09 07:56:31,551 INFO  [io.qua.qui.dep.pac.PackageManagerRunner] (build-3) Running Quinoa package manager live coding as a dev service: npm run start
2024-01-09 07:56:37,934 INFO  [io.qua.qui.dep.ForwardedDevProcessor] (build-3) Quinoa package manager live coding is up and running on port: 3003 (in 6383ms)
2024-01-09 07:56:37,937 INFO  [io.qua.qui.dep.ForwardedDevProcessor] (build-48) Quinoa is forwarding unhandled requests to port: 3003

```

**Reply by ia3andy** [reply]
Are you using RESTEasy Classic or Reactive for the api?

**Reply by ia3andy** [reply]
Would you try to remove most of the logic of your app and provide a reproducer?

**Reply by mvolf** [reply]
We're using RESTEasy Reactive (quarkus-resteasy-reactive-jackson). It would be quite time consuming to provide you with a reproducer, since it's quite a large project and removing the logic is not trivial.

**Reply by mvolf** [reply]
I've noticed that the first two lines from the log while starting the app (I posted above) are not present in the logs while using 1.2.4 version. With the old version I can only see the following:

```
2024-01-09 14:11:54,688 INFO  [io.qua.qui.dep.ForwardedDevProcessor] (build-2) Quinoa package manager live coding is up and running on port: 3003 (in 8011ms)
2024-01-09 14:11:54,691 INFO  [io.qua.qui.dep.ForwardedDevProcessor] (build-80) Quinoa is forwarding unhandled requests to port: 3003
```
Not sure if the log verbosity changed between the versions or it could point to a culprit to my issue.

**Reply by ia3andy** [reply]
Could you post non sensitive stuff from your application.properties?
Are you using SPA routing? is your API using a specific prefix?

**Reply by mvolf** [reply]
SPA routing is enabled in the properties, but disabling it doesn't help. All APIs have the /api/ prefix, except the public ones which start with /public/.
Here are the Quarkus/Quinoa related properties:

```
quarkus.http.auth.permission.authenticated.paths=/api/*
quarkus.http.auth.permission.authenticated.policy=authenticated

quarkus.http.auth.permission.public.paths=/public/*,/static/*,/*.map,/*.css,/*.js,/*.png,/*.html,/*.woff2,/*.ico,/robots.txt
quarkus.http.auth.permission.public.policy=permit
quarkus.http.auth.permission.public.methods=GET,HEAD,POST

quarkus.quinoa.enable-spa-routing=true
quarkus.http.enable-compression=true
quarkus.log.level=INFO
quarkus.banner.enabled=false
quarkus.package.type=uber-jar
quarkus.container-image.builder=docker

quarkus.qute.strict-rendering=false
quarkus.qute.property-not-found-strategy=output-original

quarkus.health.extensions.enabled=false

quarkus.kubernetes-client.connection-timeout=2000
quarkus.kubernetes-client.request-timeout=10000
quarkus.kubernetes-client.devservices.enabled=false

quarkus.swagger-ui.always-include=true

quarkus.quinoa.ci=false
quarkus.quinoa.package-manager-install=false
quarkus.quinoa.package-manager-install.node-version=18.12.1
quarkus.quinoa.dev-server.port=3003
quarkus.quinoa.dev-server.check-timeout=60000

quarkus.tls.trust-all=true
```

**Reply by ia3andy** [reply, marked as answer]
Could you try to add this and let me know:
```
quarkus.quinoa.ignored-path-prefixes=/api,/q,/public
```

Add any path prefix that is non UI.

**Reply by ia3andy** [reply]
My guess is that this is what changed the behavior, it's hard to know for sure:
https://github.com/quarkiverse/quarkus-quinoa/commit/51c055e4d006aa824569d3980300665e07264615#diff-dccbd15f74fe7c0dceaef61b9cd9764556b87260be3ab7e4163e757a919c9131L88-L92



But avoiding to process undesired requests through Quinoa is the best way forward.

**Reply by mvolf** [reply]
Adding the ignored path prefixes helped! Thanks @ia3andy !
